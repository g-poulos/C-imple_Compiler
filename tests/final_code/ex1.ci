program main
{
    declare a,b,c;
    function f(in a, inout b)
    {
        b:=a+1;
        c:=4;
        return(b);
    }

    a:=1;
    c:=f(in a, inout b);
    print(c);
    print(b);
}.