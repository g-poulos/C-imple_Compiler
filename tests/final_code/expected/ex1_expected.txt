L0:
b Lmain
L1:
sw ra, -0(sp)
L2:
lw t1, -12(sp)
li t2, 1
add t1, t1, t2
sw t1, -20(sp)
L3:
lw t1, -20(sp)
lw t0, -16(sp)
sw t1, (t0)
L4:
li t1, 4
sw t1, -20(gp)
L5:
lw t0, -16(sp)
lw t1, (t0)
lw t0, -8(sp)
sw t1, (t0)
L6:
lw ra, -0(sp)
jr ra
Lmain:
L7:
addi sp, sp, 28
move gp, sp
L8:
li t1, 1
sw t1, -12(gp)
L9:
addi fp, sp, 24
lw t1, -12(gp)
sw t1, -12(fp)
L10:
addi t0, sp, -16
sw t0, -16(fp)
L11:
addi t0, sp, -24
sw t0, -8(fp)
L12:
sw sp, -4(fp)
addi sp, sp, 24
jal L1
addi sp, sp, -24
L13:
lw t1, -24(sp)
sw t1, -20(sp)
L14:
lw t1, -20(sp)
L15:
lw t1, -16(sp)
L16:
li a0, 0
li a7,93
ecall
L17:
lw ra, -0(sp)
